contextutil/context.go

```diff
@@ -0,0 +1,35 @@
+package contextutil
```

Comment:
I believe adding upper layer packages dependencies here will lead to an overhead of cyclic dependencies. Instead of having each package "know what look for". 
We can discuss more about it next week.

Reply:
yes, Im already away of this issue, particular the identity package has an issue, because of the use of identity here. My idea is we should use only the ID bytes in the context.